packages/network-controller/src/NetworkController.test.ts

```diff
@@ -1,224 +1,3653 @@
+//                                                 lookupNetwork
+
+describe('NetworkController', () => {
   beforeEach(() => {
-    messenger = new ControllerMessenger().getRestricted({
-      name: 'NetworkController',
-      allowedEvents: ['NetworkController:providerConfigChange'],
-      allowedActions: [],
-    });
+    jest.resetAllMocks();
+    jest.useFakeTimers();
   });
 
   afterEach(() => {
-    sinon.restore();
+    jest.useRealTimers();
   });
 
-  it('should set default state', () => {
-    const controller = new NetworkController({
-      messenger,
-      infuraProjectId: 'potate',
+  describe('constructor', () => {
+    it('initializes the state with some defaults', async () => {
+      await withController(({ controller }) => {
+        expect(controller.state).toStrictEqual({
+          network: 'loading',
+          isCustomNetwork: false,
+          providerConfig: { type: 'mainnet' as const, chainId: '1' },
+          properties: { isEIP1559Compatible: false },
+        });
+      });
     });
 
-    expect(controller.state).toStrictEqual({
-      network: 'loading',
-      isCustomNetwork: false,
-      properties: { isEIP1559Compatible: false },
-      providerConfig: {
-        type: 'mainnet',
-        chainId: '1',
-      },
+    it('merges the given state into the default state', async () => {
+      await withController(
+        {
+          state: {
+            isCustomNetwork: true,
+            properties: { isEIP1559Compatible: true },
+          },
+        },
+        ({ controller }) => {
+          expect(controller.state).toStrictEqual({
+            network: 'loading',
+            isCustomNetwork: true,
+            providerConfig: { type: 'mainnet', chainId: '1' },
+            properties: { isEIP1559Compatible: true },
+          });
+        },
+      );
     });
   });
 
-  it('should create a provider instance for default infura network', () => {
-    const networkControllerOpts = {
-      infuraProjectId: 'foo',
-      messenger,
-    };
-    const controller = new NetworkController(networkControllerOpts);
-    controller.providerConfig = {} as ProviderConfig;
-    expect(controller.provider instanceof Web3ProviderEngine).toBe(true);
-  });
+  describe('providerConfig property', () => {
+    describe('get', () => {
+      it('throws', async () => {
+        await withController(({ controller }) => {
+          expect(() => controller.providerConfig).toThrow(
+            'Property only used for setting',
+          );
+        });
+      });
+    });
+
+    describe('set', () => {
+      (['kovan', 'mainnet', 'rinkeby', 'goerli', 'ropsten'] as const).forEach(
+        (networkType) => {
+          describe(`when the provider config in state contains a network type of "${networkType}"`, () => {
+            ['1', '3', '4', '5', '42', ''].forEach((chainId) => {
+              describe(`when the provider config in state contains a chain ID of "${chainId}"`, () => {
```

Comment:
This `describe` block and the next one seem to be at the wrong indentation level. They're scoped under the `networkType` loop, but they don't use the network type at all.

Reply:
True! The network type doesn't matter for these tests. The same goes for the `getEIP1559Compatibility` call, I noticed — that happens regardless of the network type or chain ID. Moved out in [bfaf5c7](https://github.com/MetaMask/core/pull/1039/commits/bfaf5c784b3617c54f58735088e9e5ba54551d86).